addons/atlassian-jira/uninstall.js

```diff
@@ -3,16 +3,22 @@ This is the uninstallation logic of the Lifecycle Manager.
 */
 
 const Sdk = require('@fusebit/add-on-sdk');
-const Superagent = require('superagent');
 
 module.exports = async (ctx) => {
-  let functionCtx = await Sdk.getFunctionDefinition(ctx);
+  if (!ctx.caller.permissions) {
+    return { status: 403 };
+  }
 
-  await Sdk.deleteStorage(ctx, functionCtx.configuration);
-  Sdk.debug('Storage deleted', functionCtx.location);
+  const storage = await Sdk.createStorageClient(
```

Comment:
Try/catch around storage so that deleteFunction will still get called even if storage glitches.

Reply:
Both delete storage and delete function report success if the storage/function does not exist. The idea behind this logic in uninstall was for it to be eventually consistent - if uninstall fails, you call it again until you get a success. The only reason deleting storage or a function should fail consistently if we have a bug in our code, in which case hiding it with a try/catch is only going to make the situation less consistent and harder to diagnose over time.